content/articles/ssh-tunneling/index.md

```diff
@@ -0,0 +1,130 @@
+
+When our public SSH server receives this, it acknowledges that our local machine wants to access resources at IP address `192.168.1.3` that it has access to. So it will forward the request internally because they are on the same network. The server will receive the request, get the response, and then forward it back to our local machine. With that, we have accessed our remote resources.
+
+In some instances,  let's say our remote IP address is forbidden from our local network i.e, our ISP is blocking access to specific domains, we can tunnel it through our secure SSH and then make our public SSH server request on our behalf.
+
+### Remote port forwarding
+Remote port forwarding is when we want other people to have access to our local resources. For instance, let's say we have a web server on our machines, and we want to send a URL for someone to test our application. What do we do? We still use our public SSH server as a centralized location and ask the public server to listen to a port.  Any traffic that goes through that port is then directed to us. That is, it will be forwarded to us.
+
+Let us have a look at the image below:
+
+![Remote port forwarding](/engineering-education/ssh-tunneling/remote.jpg)
+
+From the image above, we have our:
+- Internal resources on IP address `10.0.0.3` on port 8080 that we want people to have access to.
+- Machine of IP address `10.0.0.4` on the same network as our internal resources as an SSH client.
+- Public SSH server and a remote machine on the same network.
+
+Let us have a look at the command below:
+
+Remote port forwarding syntax:
+
+```bash
+ssh -R <remote port>:<local ip address>:<local port> <remote ip address>
+```
+
+```bash
+ssh -R 8888:10.0.0.3:8080 44.11.22.33
+```
+
+The command above is executed on our local machine, that is, `10.0.0.4`. 8888 is our remote port. So, the public server that we are accessing will listen on port 8888. Gateway configurations needs to be enabled on our local machine to enable remote port forwarding.
+
+The main idea here is that we need anything that goes to port 8888 to be forwarded to the local machine that we have on `10.0.0.3` on port 8080. What we will do is to establish a TCP connection with the public SSH server and our local machine. 
+
+The server will listen to port 8888 and if there is a TCP request on 44.11.22.33 on the same port, the SSH server will take that packet, encapsulate it a legitimate SSH request. It will then send it over through port 22 since it is encrypted and no one on the network can access it.
+
+We will receive the request on our local device, as the ssh client at 10.0.0.4 will have the ability to know that it is a tunnel. It will make the request to the internal resource on 10.0.0.3 on the behalf of our public SSH server and access the internal resource. Afterwards, it will send it back to our public ssh server.
+
```

Comment:
```suggestion
The SSH server will listen to port 8888 for a TCP request on `44.11.22.33` on the same port. The SSH server will take that packet and encapsulate it as a legitimate SSH request. It will then send it over through port 22 since it is encrypted and no one on the network can access it.

We will receive the request on our local device as the SSH client at `10.0.0.4` will have the ability to know that it is a tunnel. It will request the internal resource on `10.0.0.`3 on behalf of our public SSH server and access the internal resource. Afterwards, it will send it back to our public ssh server.
```